### PR TITLE
Pnpm instead of yarn

### LINKED ISSUE #210: Consider `pnpm` instead of `yarn`
I've seen some big projects switch to `pnpm`, like next and vite. It's faster and re-uses dependencies on your disk (although yarn has an option for that, too). Even unoptimized it's [about as fast](https://divriots.com/blog/switching-to-pnpm) as yarn 3 + cache (which is what we use).

I also had some problems with running scripts from sub projects in workspaces (e.g. running `yarn test` from `/react` required me to duplicate the dependency. Not sure if `pnpm` fixes this, but it's worth trying.

### LINKED ISSUE #213: Hierarchy / breadcrumbs should always be visible
Currently, many `Page` views have a `<Parent>` element at the top, showing breadcrumbs, giving the user an understanding of the hierarchy of the resource.

This is often very useful information. However, it is not _always_ shown. In the `Document` or `ChatRoom` pages, for example, we don't render the `<Parent>`.

We need more consistency, but I don't think rendering the parent everywhere is the solution. I think we need some UI element that is always visible. Some ideas:

- Always show the parent at the top.
- Always show a `to parent` button in the navigation bar. This should probably be an up arrow or something similar. This has minimal UI impact, and it seems pretty intuitive, but it does not actually provide any information.
- Show the `parent` in the navigation bar, instead of the URL. The URL is kind of geeky, an not always relevant. We might be able to show all parents as pills in the navbar. On small screens we show the last parent, and hide the top ones (making them overflow).

### PR DESCRIPTION
**Pull request by joepio**
PR Checklist:

- [x] Link to related issue: closes #210 closes #213
- [x] Add changelog entry linking to issue
- [ ] Added tests (if needed)
- [x] (If new feature) added in description / readme

@Polleps can you try running this locally and see if that works well?

### CODE REVIEW

File: data-browser/src/components/NewInstanceButton/Base.tsx [hunk 1]
```diff
@@ -42,7 +44,7 @@ export function Base({
       subtle={subtle}
       title={agent ? `Create a new ${title}` : 'No User set - sign in first'}
     >
-      {icon ? <FaPlus /> : title}
+      {icon ? <FaPlus /> : label || title}
```
Comment (Polleps) [inline data-browser/src/components/NewInstanceButton/Base.tsx:47]: Prefere `??` over `||` in this case as `||` does type coercion

File: data-browser/src/components/SideBar/ResourceSideBar/ResourceSideBar.tsx [hunk 2]
```diff
@@ -139,6 +140,6 @@ const Title = styled(AtomicLink)<TitleProps>`
     cursor: pointer;
     color: ${({ active, theme: { colors } }) =>
       active ? colors.main : colors.textLight};
-    padding-left: 0px;
+    /* padding-left: 0px; */
```
Comment (Polleps) [inline data-browser/src/components/SideBar/ResourceSideBar/ResourceSideBar.tsx]: Remove line

File: data-browser/src/components/SideBar/SideBarItem.ts [hunk 3]
```diff
@@ -7,16 +7,19 @@ export interface SideBarItemProps {
 /** SideBarItem should probably be wrapped in an AtomicLink for optimal behavior */
 // eslint-disable-next-line prettier/prettier
 export const SideBarItem = styled('span')<SideBarItemProps>`
-  padding-left: ${props => props.theme.margin}rem;
-  padding-right: ${props => props.theme.margin}rem;
+  /* padding-left: ${props => props.theme.margin}rem;
```
Comment (Polleps) [inline data-browser/src/components/SideBar/SideBarItem.ts:10]: Remove lines

File: data-browser/src/components/forms/hooks/useSaveResource.ts [hunk 4]
```diff
@@ -29,6 +29,10 @@ export const useSaveResource = (
       setErr(null);
 
       try {
+        if (resource.new) {
```
Comment (Polleps) [inline data-browser/src/components/forms/hooks/useSaveResource.ts:32]: Should maybe be moved to after the save as we probably don't want to notify when the save fails.

File: data-browser/src/components/Parent.tsx [hunk 5]
```diff
@@ -39,31 +63,85 @@ type NestedParentProps = {
 function NestedParent({ subject, depth }: NestedParentProps): JSX.Element {
   const resource = useResource(subject, { allowIncomplete: true });
```
Comment (Polleps) [inline data-browser/src/components/Parent.tsx:64]: Shadowing. Instead maybe try`<NestedParent subject={parent} depth={depth + 1} />`

File: data-browser/src/components/Parent.tsx [hunk 6]
```diff
@@ -39,31 +63,85 @@ type NestedParentProps = {
...
   const [parent] = useString(resource, properties.parent);
-
+  const navigate = useNavigate();
+  const [title] = useTitle(resource);
   depth = depth + 1;
 
   // Prevent infinite recursion, set a limit to parent breadcrumbs
   if (depth > 5) {
```
Comment (Polleps) [inline data-browser/src/components/Parent.tsx:71]: Extracting this (magic) number to a constant (eg. MAX_BREADCRUM_AMOUNT) makes the code speak for itself and probably removes the need for the comment.
```ts
const MAX_BREADCRUM_AMOUNT = 4;
...
if (depth > MAX_BREADCRUM_AMOUNT) {
  ...
}
```

File: data-browser/src/components/Parent.tsx [hunk 7]
```diff
@@ -39,31 +63,85 @@ type NestedParentProps = {
...
+  }
+`;
+
+const Breadcrumb = styled.a<BreadCrumbProps>`
+  ${BreadCrumbBase}
   white-space: nowrap;
   overflow: hidden;
   text-overflow: ellipsis;
   align-self: center;
+  cursor: ${p => (p.isCurrent ? 'initial' : 'pointer')};
```
Comment (Polleps) [inline data-browser/src/components/Parent.tsx:133]: `isCurrent` doesn't seem to be used anywhere.

File: data-browser/src/views/ErrorPage.tsx [hunk 8]
```diff
@@ -60,3 +61,42 @@ function ErrorPage({ resource }: ResourcePageProps): JSX.Element {
...
+  }
+
+  public static getDerivedStateFromError(error) {
+    // Update state so the next render will show the fallback UI.
+    return { error };
+  }
+
+  public render() {
+    if (this.state.error) {
+      // You can render any custom fallback UI    w
```
Comment (Polleps) [inline data-browser/src/views/ErrorPage.tsx:90]: Looks like a comment left over from pasting.

File: data-browser/src/views/ErrorPage.tsx [hunk 9]
```diff
@@ -60,3 +61,42 @@ function ErrorPage({ resource }: ResourcePageProps): JSX.Element {
...
+      return (
+        <CrashPage
+          error={this.state.error}
+          clearError={() => this.setState(() => ({ error: undefined }))}
```
Comment (Polleps) [inline data-browser/src/views/ErrorPage.tsx:94]: Like useState the function argument version of setState is usually only for when you need to determine the state based on the previous state, meaning `setState({ error: undefined })` is enough in this case.

File: data-browser/src/views/ImporterPage.tsx [hunk 10]
```diff
@@ -28,7 +28,6 @@ export function ImporterPage({ resource }: ResourcePageProps) {
 
   return (
     <ContainerNarrow about={resource.getSubject()}>
```
Comment (Polleps) [inline data-browser/src/views/ImporterPage.tsx:30]: Forgot to remove the import

File: data-browser/src/views/ResourcePageDefault.tsx [hunk 11]
```diff
@@ -44,7 +44,6 @@ export function ResourcePageDefault({
 
   return (
     <ContainerNarrow about={resource.getSubject()}>
-      <Parent resource={resource} />
       <EditableTitle resource={resource} />
```
Comment (Polleps) [inline data-browser/src/views/ResourcePageDefault.tsx:47]: Forgot to remove import

File: react/src/hooks.ts [hunk 12]
```diff
@@ -357,40 +357,60 @@ export function useSubject(
...
+    titleHookOpts,
+  );
+  const [filename, setFileName] = useString(
+    resource,
+    urls.properties.file.filename,
+    titleHookOpts,
+  );
+
   // TODO: truncate non urls
   truncateLength = truncateLength ? truncateLength : 40;
```
Comment (Polleps) [inline react/src/hooks.ts:389]: Shadowing
```ts
function useTitle(
  resource: Resource,
  truncateLength = 40,
)
```

File: package.json [hunk 13]
```diff
@@ -2,56 +2,54 @@
...
     "typedoc": "typedoc . --options typedoc.json"
   },
   "workspaces": {
     "packages": [
       "lib",
       "react",
       "data-browser"
-    ]
+    ],
+    "packageManager": "pnpm@7.11.0"
```
Comment (Polleps) [inline package.json:51]: I think this should be at the root. Corepack does not recogonise it here but does when located at the root.

**Comment by joepio** [comment]
Tests succeeding locally, merging.

**Pull request merged**